webapp/src/ts/services/format-data-record.service.ts

```diff
@@ -689,27 +680,39 @@ export class FormatDataRecordService {
   }
 
   format(doc) {
-    const promises = [this.settingsService.get(), this.languageService.get()];
-    const patientId = doc.patient_id || (doc.fields && doc.fields.patient_id);
-    if (doc.scheduled_tasks && patientId) {
-      promises.push(this.getPatient(patientId));
-      promises.push(this.getRegistrations(patientId));
-    }
-    return Promise.all(promises).then((results) => {
-      const settings = results[0];
-      const language = results[1];
-      const context:any = {};
-      if (results.length === 4) {
-        context.patient = results[2];
-        context.registrations = (<[]>results[3]).filter((registration) => {
-          return registrationUtils.isValidRegistration(
-            registration,
-            settings
-          );
-        });
-      }
-      return this.makeDataRecordReadable(doc, settings, language, context);
-    });
+    return this.ngZone.runOutsideAngular(() => this._format(doc));
```

Comment:
Running code outside Angular makes promise resolutions not trigger change detection. We have a series of promises that we chain here: get settings, get language, get 2x registrations. Their results are intermediary, they're not affecting our view, but them resolving will trigger change detection because that's how Zone.js works. 
By running them outside Angular, we insure change detection only runs for the "final" result - when we come back to Angular.